### PR TITLE
Commit tracker

### PR DESCRIPTION
**Pull request by CermakM**
Implementation of commit tracker

ref: https://github.com/openshiftio/openshift.io/issues/3183

### CODE REVIEW

**Comment by CermakM** [comment]
[test]

**Comment by CermakM** [comment]
[test]

**Comment by tisnik** [comment]
[test]

**Comment by tisnik** [comment]
[test]

File: src/toolkit/preprocessing/ecos.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        """Return string representation of the current instance."""
+        return "[{eco}]: {proj}, version {ver}".format(
+            eco=self.ecosystem,
+            proj=self._name,
+            ver=self._version
+        )
+
+    def __hash__(self):
+        """Return hash of the current instance."""
+        return hash(str(self))
```
[the remark this answers is not in the document]
Comment (tisnik) [reply src/toolkit/preprocessing/ecos.py]: @msrb @CermakM this seems to be ok there 'cause the `str` is reimplemented as well and it basically returns E/P/V that has to be unique ;)

File: src/toolkit/preprocessing/ecos.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    def __eq__(self, other):
+        """Compare two package objects for equality."""
+        return hash(other) == hash(self)
```
[the remark this answers is not in the document]
Comment (tisnik) [reply src/toolkit/preprocessing/ecos.py]: actually why not to compare str() output directly?

**Comment by CermakM** [comment]
@msrb I have no idea how that happened, but the word `print` somehow disappeared .. will be fixed in no time.

File: src/toolkit/preprocessing/ecos.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (msrb) [reply src/toolkit/preprocessing/ecos.py]: Ah, right, I overlooked the __str__ implementation. Makes sense now, thanks :wink:

**Pull request merged**